lib/seahorn/BvOpSem2MemRepr.cc

```diff
@@ -13,7 +13,20 @@ auto as_std_array(const T &t, const Rest &... rest) ->
 
 namespace seahorn {
 namespace details {
-
+auto addAlignmentConstraint = [](auto &m_ctx, Expr len, unsigned lenSzInBits,
+                                 unsigned wordSzInBytes) {
+  // This function adds an assumption that (len % wordSzInBytes) == 0
+  // TODO: use shl isntead of urem once alu api supports it.
+  Expr remainder = m_ctx.alu().doURem(
+      len, m_ctx.alu().ui(wordSzInBytes, lenSzInBits), lenSzInBits);
+  Expr lenIsAligned =
+      m_ctx.alu().doEq(remainder, m_ctx.alu().ui(0, lenSzInBits), lenSzInBits);
+  m_ctx.addScopedRely(lenIsAligned);
+  LOG(
+      "opsem", if (dagSize(lenIsAligned) < 64) {
```

Comment:
we should always print a message that some extra constraints like alignment are being forcefully added.
However, we do not wish to print very large expressios.

The current code does not print anything when the expression is too big.

Also the fact that `64` is **too big** should be factored out into some definition so that we can change it in one place.